contrib/scalacheck/shared/src/test/scala/eu/timepit/refined/scalacheck/NumericArbitrarySpec.scala

```diff
@@ -17,24 +20,45 @@ class NumericArbitrarySpec extends Properties("NumericArbitrary") {
   property("Less.negative") =
     checkArbitraryRefType[Refined, Int, Less[W.`-100`.T]]
 
+  property("Less.natural") =
+    checkArbitraryRefType[Refined, Long, Less[Nat._10]]
+
   property("LessEqual.positive") =
     checkArbitraryRefType[Refined, Int, LessEqual[W.`42`.T]]
 
   property("LessEqual.negative") =
     checkArbitraryRefType[Refined, Int, LessEqual[W.`-42`.T]]
 
+  property("LessEqual.natural") =
+    checkArbitraryRefType[Refined, Long, LessEqual[Nat._10]]
+
   property("Greater.positive") =
     checkArbitraryRefType[Refined, Int, Greater[W.`100`.T]]
 
   property("Greater.negative") =
     checkArbitraryRefType[Refined, Int, Greater[W.`-100`.T]]
 
+  property("Greater.natural") =
+    checkArbitraryRefType[Refined, Long, Greater[Nat._10]]
+
+  property("Greater range") =
+    forAll { (min: Refined[Long, Greater[Nat._10]], max: Refined[Long, Greater[Nat._10]]) =>
```

Comment:
Yup, that seems like a reasonable hypothesis. Can probably add a precondition on the outer property.